components/definition/stimulation/NetPyNEStimulationSource.js

```diff
@@ -10,79 +10,259 @@ import RaisedButton from 'material-ui/RaisedButton';
+          <NetPyNEField id="netParams.stimSourceParams.i">
+            <PythonControlledTextField
+              model={"netParams.stimSourceParams['" + this.props.name + "']['i']"}
+            />
+          </NetPyNEField>
+        </div>
+    } else if (this.state.sourceType=='AlphaSynapse'){
+      var variableContent = 
+        <div>
+          <NetPyNEField id="netParams.stimSourceParams.onset">
+            <PythonControlledTextField
+              model={"netParams.stimSourceParams['" + this.props.name + "']['onset']"}
+            />
+          </NetPyNEField>
+          <NetPyNEField id="netParams.stimSourceParams.tau">
+            <PythonControlledTextField
+              model={"netParams.stimSourceParams['" + this.props.name + "']['tau']"}
+            />
+          </NetPyNEField>
+          <NetPyNEField id="netParams.stimSourceParams.gmax">
+            <PythonControlledTextField
+              model={"netParams.stimSourceParams['" + this.props.name + "']['gmax']"}
+            />
+          </NetPyNEField>
+          <NetPyNEField id="netParams.stimSourceParams.e">
+            <PythonControlledTextField
+              model={"netParams.stimSourceParams['" + this.props.name + "']['e']"}
+            />
+          </NetPyNEField>
+          <NetPyNEField id="netParams.stimSourceParams.i">
+            <PythonControlledTextField
+              model={"netParams.stimSourceParams['" + this.props.name + "']['i']"}
+            />
+          </NetPyNEField>
+        </div>
+    }else if (this.state.sourceType=='NetStim'){
+      var variableContent = 
+        <div>
+          <NetPyNEField id="netParams.stimSourceParams.interval">
+            <PythonControlledTextField
+              model={"netParams.stimSourceParams['" + this.props.name + "']['interval']"}
+            />
+          </NetPyNEField>
+          <NetPyNEField id="netParams.stimSourceParams.number">
+            <PythonControlledTextField
+              model={"netParams.stimSourceParams['" + this.props.name + "']['number']"}
+            />
+          </NetPyNEField>
+          <NetPyNEField id="netParams.stimSourceParams.start">
+            <PythonControlledTextField
+              model={"netParams.stimSourceParams['" + this.props.name + "']['start']"}
+            />
+          </NetPyNEField>
+          <NetPyNEField id="netParams.stimSourceParams.noise">
+            <PythonControlledTextField
+              model={"netParams.stimSourceParams['" + this.props.name + "']['noise']"}
+            />
+          </NetPyNEField>
+        </div>
+    } else if (this.state.sourceType=='SEClamp'){
+      var variableContent = 
+        <div>
+          <NetPyNEField id="netParams.stimSourceParams.rs">
+            <PythonControlledTextField
+              model={"netParams.stimSourceParams['" + this.props.name + "']['rs']"}
+            />
+          </NetPyNEField>
+          <NetPyNEField id="netParams.stimSourceParams.dur" className="listStyle">
```

Comment:
Same id for two different fields

Reply:
@rodriguez-facundo I saw that you changed the id for all the components in the file. I just meant two components with the same id "netParams.stimSourceParams.dur". Probably it is better if you revert that change. metadata.py uses the ids to extract the help, hint, type, etc.